apps/mobile/src/hooks/useDebounce.ts

```diff
@@ -0,0 +1,19 @@
+import { useEffect, useState } from 'react';
+
+function useDebounce<T>(value: T, delay: number): T {
+  const [debouncedValue, setDebouncedValue] = useState(value);
+
+  useEffect(() => {
+    const handler = setTimeout(() => {
+      setDebouncedValue(value);
+    }, delay);
+
+    return () => {
+      clearTimeout(handler);
+    };
+  }, [delay, value]);
+
+  return debouncedValue;
+}
+
+export default useDebounce;
```

Comment:
@tbezman I copy the whole thing from `/web` project. Should we put it as shared library?